### ISSUE TITLE
Package installed in infinite loop

### ISSUE THREAD

**Issue by dobesv** [parcel-bundler/parcel#1187]
🐛 bug report

A particular package causes Parcel to endless attempt to install it over and over without cease.

### 🎛 Configuration (.babelrc, package.json, cli command)

```
$ cat index.html
<script src="index.js"></script>

$ cat index.js

import 'waypoints';

$ cat package.json
{
  "name": "temp",
  "version": "1.0.0",
  "main": "index.js",
  "license": "MIT",
  "dependencies": {
    "waypoints": "^4.0.1"
  }
}

$ FORCE_COLOR=0 parcel watch index.html
⏳  Building...

⏳  Building index.html...
⏳  Building index.js...
⏳  Installing waypoints...
yarn add v1.5.1
warning ../package.json: No license field
[1/4] Resolving packages...
[2/4] Fetching packages...
[3/4] Linking dependencies...
[4/4] Building fresh packages...
success Saved 1 new dependency.
info Direct dependencies
└─ waypoints@4.0.1
info All dependencies
└─ waypoints@4.0.1
⏳  Building package.json...
⏳  Building index.js...
Done in 0.40s.
🚨  /home/ubuntu/temp/index.js:2:7: Cannot find module 'waypoints' from '/home/ubuntu/temp'
  1 | 
> 2 | import 'waypoints';
    |        ^
  3 | 
  4 | 
⏳  Building...
⏳  Installing waypoints...
yarn add v1.5.1
warning ../package.json: No license field
[1/4] Resolving packages...
[2/4] Fetching packages...
[3/4] Linking dependencies...
[4/4] Building fresh packages...
success Saved 1 new dependency.
info Direct dependencies
└─ waypoints@4.0.1
info All dependencies
└─ waypoints@4.0.1
⏳  Building package.json...
Done in 0.36s.
🚨  /home/ubuntu/temp/index.js:2:7: Cannot find module 'waypoints' from '/home/ubuntu/temp'
  1 | 
> 2 | import 'waypoints';
    |        ^
  3 | 
  4 | 
⏳  Building...
⏳  Building index.js...
⏳  Installing waypoints...

^C

```

**Comment by devongovett** [comment]
cc. @davidnagli

**Comment by DeMoorJasper** [comment]
Could you try again with parcel 1.7.1 it might be related to running installs inside workers instead of master thread. This got fixed in 1.7.1

**Comment by dobesv** [comment]
```
$ parcel --version
1.7.1
$ parcel index.html | cat
Server running at http://localhost:1234 
⏳  Building...

⏳  Building index.html...
⏳  Building index.js...
⏳  Building re-tree.js...
⏳  Building index.js...
⏳  Building ua-device-detector.js...
⏳  Installing waypoints...
yarn add v1.5.1
warning ../package.json: No license field
[1/4] Resolving packages...
[2/4] Fetching packages...
[3/4] Linking dependencies...
[4/4] Building fresh packages...
success Saved 1 new dependency.
info Direct dependencies
└─ waypoints@4.0.1
info All dependencies
└─ waypoints@4.0.1
⏳  Building package.json...
⏳  Building index.js...
⏳  Building index.js...
⏳  Building re-tree.js...
⏳  Building ua-device-detector.js...
Done in 5.41s.
🚨  /home/ubuntu/temp/index.js:4:7: Cannot find module 'waypoints' from '/home/ubuntu/temp'
  2 | import 're-tree';
  3 | import 'ua-device-detector';
> 4 | import 'waypoints';
    |        ^
  5 | 
  6 | 
⏳  Building...
⏳  Installing waypoints...
yarn add v1.5.1
warning ../package.json: No license field
[1/4] Resolving packages...
```

One thing is, this waypoints library is a bit weird, which is probably part of the cause here.  If I change the way that I import waypoints then the problem goes away.  I just reported this because I thought it might indicate some underlying problem that needs to be addressed.

**Comment by DeMoorJasper** [comment]
Seems like it just fails to install and reinstalls again

**Comment by dobesv** [comment]
@DeMoorJasper I printed the three files required in the original report, just create a folder with those three files and you're good to go.

The module is installed into node_modules, but Parcel doesn't think it is really installed.  I think that's because `require('waypoints')` doesn't actually work for this module.  It's kind of a weird scenario.  But it indicates that Parcel's way of determining whether the package installed isn't quite perfect.

I imagine you might have a similar problem at time for a package that legitimately doesn't exist or fails to install for some other reason.  Perhaps Parcel watch needs to "remember" which it has already tried to auto-install and not try again.

**Comment by DeMoorJasper** [comment]
Sounds like a great suggestion @davidnagli have any time to look into this?

**Comment by devongovett** [comment]
Is this fixed by #1241 @DeMoorJasper?

**Comment by DeMoorJasper** [comment]
@devongovett I think it's fair to say it's installed as it apparently didn't had a main but a folder, and I think that's been resolved by the resolver changes?
